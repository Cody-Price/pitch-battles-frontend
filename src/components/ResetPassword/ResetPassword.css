.reset-password-container-frame {
  position: absolute;
  width: 300px;
  height: 390px;
  left: 0;
  right: 0;
  bottom: 700px;
  transition: 0.4s;
  margin: 20px auto;
}

.reset-password-container-frame.true {
  bottom: 0;
}

.reset-password-form {
  position: relative;
}

.reset-pw-label {
  display: block;
  font-family: "chronotype";
  font-size: 1.1rem;

  width: 200px;
  height: 1rem;
  margin: 1rem auto;
}

.reset-pw-input {
  display: block;
  margin: 1rem auto;
  width: 220px;
  font-family: "chronotype";
  font-size: 1.1rem;
  padding-left: 2rem;
  padding-top: 3px;
}

.reset-pw-button,
.back-to-login-btn {
  font-family: "chronotype";
  font-size: 1.1rem;
  margin: 1rem auto;
  padding-top: 4px;
  border-radius: 1px;
}

.back-to-login-btn {
  margin-top: 9rem;
}

.password-reset-icon-one,
.password-reset-icon-two {
  position: absolute;
  background: url(../Landing/images/lock.png);
  background-size: cover;
  left: 1.8rem;
  width: 13px;
  height: 16px;
  top: 2.2rem;
}

.password-reset-icon-two {
  top: 4.55rem;
}

.webtoken-icon {
  position: absolute;
  background: url(../StudentDash/images/blue-gem.png);
  background-size: cover;
  left: 1.6rem;
  width: 18px;
  height: 16px;
  top: 6.95rem;
}

@media (max-width: 700px) {
  .password-reset-icon-one,
  .password-reset-icon-two,
  .webtoken-icon {
    display: none;
  }

  .reset-pw-input {
    padding-left: 0.5rem;
    padding-top: 4px;
  }

  .back-to-login-btn {
    position: absolute;
    margin-left: auto;
    margin-right: auto;
    left: 0;
    right: 0;
    bottom: 0rem;
    width: 4rem;
    padding: 4px 0 3px;
    text-align: center;
  }
}
