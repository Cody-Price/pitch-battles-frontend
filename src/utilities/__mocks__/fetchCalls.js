import mockUser from "../../utilities/mockUser";
import mockStudentClass from "../mockStudentClass";
import { mockTeacherClass } from "../mockTeacherClass";

export const login = async body => {
  const mockResponse = { access_token: "test" };
  return Promise.resolve(mockResponse);
};

export const signUp = async body => {
  const mockResponse = { response: {} };

  return Promise.resolve(mockResponse);
};

export const userFetch = async webToken => {
  const mockResponse = mockUser;
  return Promise.resolve(mockResponse);
};

export const postGameUserUpdate = async (update, webToken) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const forgotMyPasswordCall = async email => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const changeAvatarFetch = async (avatar, id, webToken) => {
  const mockResponse = { reponse: [] };

  return Promise.resolve(mockResponse);
};

export const changeProfileFetch = async (name, id, webToken) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const changePasswordFetch = async (
  oldPassword,
  newPassword,
  id,
  webToken
) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const createClassFetch = async (name, webToken) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const joinClassFetch = async (class_key, id, webToken) => {};

export const leaveClassFetch = async (student_id, class_id, webToken) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const resetPasswordFetch = async (password, token) => {
  const mockResponse = { response: [] };

  return Promise.resolve(mockResponse);
};

export const studentClassFetch = async webToken => {
  const mockResponse = mockStudentClass;

  return Promise.resolve(mockResponse);
};

export const teacherAllClassesFetch = async webToken => {
  const mockResponse = mockTeacherClass;

  return Promise.resolve(mockResponse);
};

export const teacherSpecificClassFetch = async (id, webToken) => {
  const mockResponse = mockTeacherClass;

  return Promise.resolve(mockResponse);
};

export const deleteClassFetch = async (id, webToken) => {
  const mockResponse = { data: "success" };

  return Promise.resolve(mockResponse);
};
