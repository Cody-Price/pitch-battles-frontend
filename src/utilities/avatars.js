const avatars = [
  "one",
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight",
  "nine",
  "ten",
  "eleven",
  "twelve",
  "thirteen",
  "fourteen"
];

export default avatars;
