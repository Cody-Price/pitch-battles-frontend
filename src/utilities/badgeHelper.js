export const badgeHelper = [
  { name: "level_one", badge: "level one: completed", rank: 2 },
  { name: "level_one_perfect", badge: "level one: perfect", rank: 3 },
  { name: "level_two", badge: "level two: completed", rank: 4 },
  { name: "level_two_perfect", badge: "level two: perfect", rank: 6 },
  { name: "level_three", badge: "level three: completed", rank: 7 },
  { name: "level_three_perfect", badge: "level three: perfect", rank: 9 },
  { name: "level_four", badge: "level four: completed", rank: 11 },
  { name: "level_four_perfect", badge: "level four: perfect", rank: 12 },
  { name: "all_perfect", badge: "all levels: perfect", rank: 14 },
  { name: "five_games", badge: "play 5 games", rank: 1 },
  { name: "ten_games", badge: "play 10 games", rank: 5 },
  { name: "twenty_games", badge: "play 20 games", rank: 8 },
  { name: "fifty_games", badge: "play 50 games", rank: 10 },
  { name: "one_hundred_games", badge: "play 100 games", rank: 13 },
  { name: "two_hundred_games", badge: "play 200 games", rank: 15 },
  { name: "five_hundred_games", badge: "play 500 games", rank: 16 },
  { name: "one_thousand_games", badge: "play 1000 games", rank: 17 }
];
