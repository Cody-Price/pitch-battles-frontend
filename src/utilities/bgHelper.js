export const bgHelper = [
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight"
];
