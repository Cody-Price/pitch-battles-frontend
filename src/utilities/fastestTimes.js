export const fastestTimes = user => {
  console.log(user);
};
