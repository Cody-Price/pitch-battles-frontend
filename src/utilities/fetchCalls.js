export const login = async body => {
  const url = "https://pitch-battles-api.herokuapp.com/login";

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };

  const response = await fetch(url, options);
  return await response.json();
};

export const signUp = async body => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/users";

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };

  const response = await fetch(url, options);
  return await response.json();
};

export const userFetch = async webToken => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/dashboard/";

  const options = {
    method: "GET",
    mode: "cors",
    headers: {
      Authorization: `bearer ${webToken}`,
      "Content-Type": "application/json"
    }
  };

  console.log(options);

  const response = await fetch(url, options);
  return await response.json();
};

export const postGameUserUpdate = async (update, webToken) => {
  const url = "https://pitch-battles-api.herokuapp.com/api/v1/users";

  const body = update;

  const options = {
    body: JSON.stringify(body),
    method: "POST",
    mode: "cors",
    headers: {
      "Content-Type": "application/json"
    }
  };
  return;
};

export const forgotMyPasswordCall = async email => {
  // const url
  // return response
  return Promise.resolve(true);
};
