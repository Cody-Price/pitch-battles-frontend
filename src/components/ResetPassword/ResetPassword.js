import React, { Component } from "react";
import "./ResetPassword.css";
import PropTypes from "prop-types";

import { resetPasswordFetch } from "../../utilities/fetchCalls";

class ResetPassword extends Component {
  constructor() {
    super();

    this.state = {
      password: "",
      confirmPassword: "",
      webToken: "",
      passwordMatchError: false,
      webTokenError: false,
      success: false
    };
  }

  componentWillUnmount() {
    clearTimeout(this._timeout);
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = async event => {
    event.preventDefault();

    if (this.state.password !== this.state.confirmPassword) {
      this.setState({
        passwordMatchError: true,
        webTokenError: false,
        success: false
      });
      this._timeout = setTimeout(this.clearErrors, 5000);
      return;
    }

    try {
      const data = await resetPasswordFetch(
        this.state.password,
        this.state.webToken
      );
      if (data.error) {
        throw data.error;
      } else {
        this.setState({
          webTokenError: false,
          passwordMatchError: false,
          success: true
        });
      }
      console.log(data);
    } catch (error) {
      this.setState({
        passwordMatchError: false,
        webTokenError: true,
        success: false
      });
      console.log(error);
      this._timeout = setTimeout(this.clearErrors, 5000);
    }
  };

  clearError = () => {
    this.setState({
      passwordMatchError: false,
      webTokenError: false
    });
  };

  render() {
    return (
      <article
        className={`reset-password-container-frame ${this.props.status}`}
      >
        <form
          onSubmit={e => {
            this.handleSubmit(e);
          }}
          className={`reset-password-form `}
        >
          <p className="reset-pw-label">reset password</p>
          <div className="password-reset-icon-one" />
          <input
            name="password"
            value={this.state.password}
            className="reset-pw-input pw-reset-field"
            placeholder="password"
            onChange={e => {
              this.handleChange(e);
            }}
          />
          <div className="password-reset-icon-two" />
          <input
            name="confirmPassword"
            value={this.state.confirmPassword}
            className="reset-pw-input confirm-pw-reset-field"
            placeholder="confirm password"
            onChange={e => {
              this.handleChange(e);
            }}
          />
          <div className="webtoken-icon" />
          <input
            name="webToken"
            value={this.state.webToken}
            className="reset-pw-input webtoken-pw-reset-field"
            placeholder="reset token"
            onChange={e => {
              this.handleChange(e);
            }}
          />
          <button className="reset-pw-button">reset password</button>
        </form>
        <p className={`success-message-reset ${this.state.success}`}>
          password succesfully changed!
        </p>
        <p className={`fail-message-reset ${this.state.webTokenError}`}>
          attempt failed - check your webtoken!
        </p>
        <p className={`no-pw-match-reset ${this.state.passwordMatchError}`}>
          passwords do not match!
        </p>

        <button onClick={this.props.toLogin} className="back-to-login-btn">
          login
        </button>
      </article>
    );
  }
}

export default ResetPassword;

ResetPassword.propTypes = {
  toLogin: PropTypes.func
};
