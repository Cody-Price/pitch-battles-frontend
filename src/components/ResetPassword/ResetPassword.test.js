import { shallow } from "enzyme";
import React from "react";
import ResetPassword from "./ResetPassword";
import { resetPasswordFetch } from "../../utilities/fetchCalls";
const Enzyme = require("enzyme");
const EnzymeAdapter = require("enzyme-adapter-react-16");

jest.mock("../../utilities/fetchCalls");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("ResetPassword", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ResetPassword />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleChange", () => {
    it("should update state", () => {
      const mockEvent = {
        target: {
          name: "password",
          value: "password"
        }
      };

      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state().password).toEqual("password");
    });

    it("should call handleChange on change", () => {
      const mockEvent = {
        target: {
          name: "password",
          value: "password"
        }
      };

      const spy = jest.spyOn(wrapper.instance(), "handleChange");

      wrapper.instance().forceUpdate();

      wrapper.find(".pw-reset-field").simulate("change", mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });

    it("should call handleChange on change", () => {
      const mockEvent = {
        target: {
          name: "confirmPassword",
          value: "password"
        }
      };

      const spy = jest.spyOn(wrapper.instance(), "handleChange");

      wrapper.instance().forceUpdate();

      wrapper.find(".confirm-pw-reset-field").simulate("change", mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });

    it("should call handleChange on change", () => {
      const mockEvent = {
        target: {
          name: "webToken",
          value: "password"
        }
      };

      const spy = jest.spyOn(wrapper.instance(), "handleChange");

      wrapper.instance().forceUpdate();

      wrapper.find(".webtoken-pw-reset-field").simulate("change", mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });
  });

  describe("handleSubmit", () => {
    it("should call event preventdefault", async () => {
      const mockPreventDefault = jest.fn();

      const mockEvent = {
        preventDefault: mockPreventDefault
      };

      await wrapper.instance().handleSubmit(mockEvent);

      expect(mockPreventDefault).toHaveBeenCalled();
    });
    it("should setstate if password and confirmpassword do no match", async () => {
      const mockPreventDefault = jest.fn();

      const mockEvent = {
        preventDefault: mockPreventDefault
      };

      wrapper.setState({
        password: 123,
        confirmPassword: 345
      });

      await wrapper.instance().handleSubmit(mockEvent);

      expect(wrapper.state().passwordMatchError).toEqual(true);
    });

    it("should call resetPasswordFetch", async () => {
      const mockPreventDefault = jest.fn();

      const mockEvent = {
        preventDefault: mockPreventDefault
      };

      wrapper.setState({
        password: 123456,
        confirmPassword: 123456
      });

      await wrapper.instance().handleSubmit(mockEvent);

      expect(resetPasswordFetch).toHaveBeenCalled();
    });
  });

  describe("clearError", () => {
    it("should setstate", () => {
      wrapper.setState({
        passwordMatchError: true,
        webTokenError: true
      });

      wrapper.instance().clearError();

      expect(wrapper.state().passwordMatchError).toEqual(false);
      expect(wrapper.state().webTokenError).toEqual(false);
    });
  });
});
