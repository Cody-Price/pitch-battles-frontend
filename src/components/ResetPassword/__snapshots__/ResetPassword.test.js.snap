// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword should match the snapshot 1`] = `
<article
  className="reset-password-container-frame undefined"
>
  <form
    className="reset-password-form "
    onSubmit={[Function]}
  >
    <p
      className="reset-pw-label"
    >
      reset password
    </p>
    <div
      className="password-reset-icon-one"
    />
    <input
      className="reset-pw-input pw-reset-field"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <div
      className="password-reset-icon-two"
    />
    <input
      className="reset-pw-input confirm-pw-reset-field"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="confirm password"
      type="password"
      value=""
    />
    <div
      className="webtoken-icon"
    />
    <input
      className="reset-pw-input webtoken-pw-reset-field"
      name="webToken"
      onChange={[Function]}
      placeholder="reset token"
      value=""
    />
    <button
      className="reset-pw-button"
    >
      reset password
    </button>
  </form>
  <p
    className="success-message-reset false"
  >
    password succesfully changed!
  </p>
  <p
    className="fail-message-reset false"
  >
    attempt failed - check your webtoken!
  </p>
  <p
    className="no-pw-match-reset false"
  >
    passwords do not match!
  </p>
  <button
    className="back-to-login-btn"
  >
    login
  </button>
</article>
`;
